tests/h/search/index_test.py

```diff
@@ -310,6 +311,66 @@ def test_annotation_is_marked_deleted(self, es_client, factories, index, search)
         assert result.get('_source').get('deleted') is True
 
 
+class TestBatchIndexer(object):
+    def test_it_indexes_all_annotations(self, db_session, es_client, factories, pyramid_request):
+        annotations = factories.Annotation.create_batch(3)
+        ids = [a.id for a in annotations]
+
+        h.search.index.BatchIndexer(db_session, es_client, pyramid_request, es_client.index).index()
+
+        for id in ids:
+            result = es_client.conn.get(index=es_client.index,
+                                        doc_type="annotation",
+                                        id=id)
+            assert result["_id"] == id
+
+    def test_it_indexes_specific_annotations(self, db_session, es_client, factories, pyramid_request):
+        annotations = factories.Annotation.create_batch(5)
+        ids = [a.id for a in annotations]
+        ids_to_index = ids[:3]
+        ids_not_to_index = ids[3:]
+
+        h.search.index.BatchIndexer(db_session, es_client, pyramid_request, es_client.index).index(ids_to_index)
+
+        for id in ids_to_index:
+            result = es_client.conn.get(index=es_client.index,
+                                        doc_type="annotation",
+                                        id=id)
+            assert result["_id"] == id
+
+        for id in ids_not_to_index:
+            with pytest.raises(elasticsearch1.exceptions.NotFoundError):
+                result = es_client.conn.get(index=es_client.index,
+                                            doc_type="annotation",
+                                            id=id)
+
+    def test_it_returns_errored_annotation_ids(self, db_session, es_client, factories, pyramid_request):
+        annotations = factories.Annotation.create_batch(3)
+        expected_errored_ids = set([annotations[0].id, annotations[2].id])
+
+        elasticsearch1.helpers.streaming_bulk = mock.Mock()
+        elasticsearch1.helpers.streaming_bulk.return_value = [(False, {'index': {'error' : 'some error', '_id': annotations[0].id}}),
```

Comment:
Some of these lines are pretty long, which makes them annoying to read in GitHub's diff viewer:

<img width="1057" alt="screenshot 2018-05-31 12 04 17" src="https://user-images.githubusercontent.com/2458/40778918-d0f2aec4-64ca-11e8-89eb-e8bcd9bd36ad.png">

Please keep them to max ~100 chars.

Reply:
Hi! yes, I have a bit of linting cleanup to do.